streams/src/test/java/org/apache/kafka/streams/kstream/internals/SessionWindowedKStreamImplTest.java

```diff
@@ -67,50 +70,84 @@ public void before() {
     }
 
     @Test
-    public void shouldCountSessionWindowed() {
-        final Map<Windowed<String>, Long> results = new HashMap<>();
+    public void shouldCountSessionWindowedWithCachingDisabled() {
```

Comment:
Extend this test for caching and non-caching (to cover `SessionTupleForwarder` and `SessionCachFlushListener` code path).

Also update this test to check the result timestamps